### PR TITLE
26 new data source name servicesldap

### LINKED ISSUE #26: [New Data Source]: name-services/ldap
### Description

Should support 
* Servers
* schema
* ad_domain
* base_dn
* bse_scope
* bind_as_cifs_servers
* preferred_ad_servers
* port
* query_timeout
* min_bind_level
* bind_dn
* bind_password
* use_start_tls
* referral_enabled
* session_security
* ldaps_enabled
* skip_config_validation

### Requested Resource(s) and/or Data Source(s)

* netapp_ontap_name-services_ldap_data_source

### Potential Terraform Configuration

_No response_

### References

https://library.netapp.com/ecmdocs/ECMLP2885799/html/#/name-services/ldap_collection_get

### Would you like to implement a fix?

None

### PR DESCRIPTION
**Pull request by chuyich**
fix #26

### CODE REVIEW

**Review by suhasbshekar — APPROVED** [review]

File: internal/provider/name_services_ldap_data_source.go [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+	CxProfileName      types.String   `tfsdk:"cx_profile_name"`
+	SVMName            types.String   `tfsdk:"svm_name"`
+	Servers            []types.String `tfsdk:"servers"`
+	Schema             types.String   `tfsdk:"schema"`
+	AdDomain           types.String   `tfsdk:"ad_domain"`
+	BaseDN             types.String   `tfsdk:"base_dn"`
+	BaseScope          types.String   `tfsdk:"base_scope"`
+	BindDN             types.String   `tfsdk:"bind_dn"`
+	BindAsCIFSServer   types.Bool     `tfsdk:"bind_as_cifs_server"`
+	PreferredADServers []types.String `tfsdk:"preferred_ad_servers"`
```
Comment (wenjun666) [inline internal/provider/name_services_ldap_data_source.go:43]: why not use types.Set?

File: internal/provider/name_services_ldap_data_source.go [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+// NameServicesLDAPDataSource defines the data source implementation.
+type NameServicesLDAPDataSource struct {
+	config resourceOrDataSourceConfig
+}
+
+// NameServicesLDAPDataSourceModel describes the data source data model.
+type NameServicesLDAPDataSourceModel struct {
+	CxProfileName      types.String   `tfsdk:"cx_profile_name"`
+	SVMName            types.String   `tfsdk:"svm_name"`
+	Servers            []types.String `tfsdk:"servers"`
```
Comment (wenjun666) [inline internal/provider/name_services_ldap_data_source.go:36]: why not use types.Set?

File: internal/provider/name_services_ldap_data_source.go (hunk 1 above)
Comment (chuyich) [reply internal/provider/name_services_ldap_data_source.go:43]: It's a list of strings. It should be straightforward with this way. Not very sure the benefit if use types.Set.

File: internal/provider/name_services_ldap_data_source.go (hunk 2 above)
Comment (chuyich) [reply internal/provider/name_services_ldap_data_source.go:36]: It's data source. And it's a list of strings. Not sure why we should use types.Set here.

**Pull request merged**